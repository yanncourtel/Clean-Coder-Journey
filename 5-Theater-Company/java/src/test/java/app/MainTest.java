package app;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

class MainTest {
//    Statement for FR_Patee
//    Hamlet: $650.00 (55 seats)
//    As You Like It: $2,140.00 (230 seats)
//    Othello: $4,120.00 (402 seats)
//    Amount owed is $7,290.05
//    You earned 643 credits

    @Test
    void invoiceFromFrCustomer() {
        // Create a stream to hold the output
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        PrintStream ps = new PrintStream(baos);
        PrintStream old = System.out;
        System.setOut(ps);

        Main.main(new String[]{"FR_Patee"});
        Approvals.verify(baos.toString());
        System.setOut(old);

    }

//    @Test
//    void invoiceFromEmptyCustomer() {
//        // Create a stream to hold the output
//        ByteArrayOutputStream baos = new ByteArrayOutputStream();
//        PrintStream ps = new PrintStream(baos);
//        PrintStream old = System.out;
//        System.setOut(ps);
//
//        Main.main(null);
//        Approvals.verify(baos.toString());
//        System.setOut(old);
//
//    }
    @Test
    void invoiceFromLuCustomer() {
        // Create a stream to hold the output
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        PrintStream ps = new PrintStream(baos);
        PrintStream old = System.out;
        System.setOut(ps);

        Main.main(new String[]{"LU_Schienergienen"});
        Approvals.verify(baos.toString());
        System.setOut(old);

    }
    @Test
    void invoiceFromBeCustomer() {
        // Create a stream to hold the output
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        PrintStream ps = new PrintStream(baos);
        PrintStream old = System.out;
        System.setOut(ps);

        Main.main(new String[]{"BE_LesFunsWallons"});
        Approvals.verify(baos.toString());
        System.setOut(old);

    }
}