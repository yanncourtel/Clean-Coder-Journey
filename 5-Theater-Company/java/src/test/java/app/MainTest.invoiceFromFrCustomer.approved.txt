Statement for FR_Patee
  Hamlet: $650.00 (55 seats)
  As You Like It: $2,140.00 (230 seats)
  Othello: $4,120.00 (402 seats)
Amount owed is $7,290.05
You earned 643 credits

